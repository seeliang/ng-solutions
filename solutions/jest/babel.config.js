module.exports = require('../jest/babel.config')