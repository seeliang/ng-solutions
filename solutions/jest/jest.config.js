module.exports = {
  preset: "jest-preset-angular",
  moduleNameMapper: {
    "\\.(jpg|jpeg|png)$": "<rootDir>/__mocks__/image.js",
  },
  setupFilesAfterEnv: [
    "<rootDir>/setupJest.ts"
  ]
}
