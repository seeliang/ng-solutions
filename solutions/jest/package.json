{
  "name": "jest",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "@babel/core": "^7.10.4",
    "@babel/preset-env": "^7.10.4",
    "@babel/preset-typescript": "^7.10.4",
    "babel-jest": "^26.1.0",
    "jest": "^26.1.0"
  }
}
