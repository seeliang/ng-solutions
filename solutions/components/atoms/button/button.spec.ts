import { async, ComponentFixture } from '@angular/core/testing';

import { configureTests, ConfigureFn } from '../../../jest/testing';

import { Button } from './index';

describe('Button', () => {
  let component: Button;
  let fixture: ComponentFixture<Button>;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [Button]
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(Button);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it('snap native with text', () => {
    component.text = "56" 
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
