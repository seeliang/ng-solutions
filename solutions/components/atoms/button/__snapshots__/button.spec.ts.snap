// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snap native 1`] = `
<div
  id="root0"
>
  <button>
    56
  </button>
</div>
`;
